nni/algorithms/compression/pytorch/pruning/structured_pruning_masker.py

```diff
@@ -477,6 +488,29 @@ def __init__(self, model, pruner, statistics_batch_num=1):
         self.pruner.iterations = 0
         self.pruner.set_wrappers_attribute("contribution", None)
         self.pruner.patch_optimizer(self.calc_contributions)
+        self.global_threshold = None
+
+    def _get_global_threshold(self):
+        channel_contribution_list = []
+        for wrapper_idx, wrapper in enumerate(self.pruner.get_modules_wrapper()):
+            channel_contribution = self.get_channel_sum(wrapper, wrapper_idx)
+            channel_contribution_list.append(channel_contribution)
+        all_channel_contributions = torch.cat(channel_contribution_list)
+        k = int(all_channel_contributions.shape[0] * self.pruner.config_list[0]['sparsity'])
```

Comment:
what if the filters' sizes are different?

Reply:
That's truly a key problem, I will fix it.

Reply:
Have fixed.

Reply:
seems we need `view(-1)` to the contribution, or if they can cat together with different size?

Reply:
That's a problem. Have fixed.